methods.rb

```diff
@@ -0,0 +1,114 @@
+# frozen_string_literal: true
+
+# methos of ruby: each, each_with_index, my_select, my_all?, my_any?, my_none?, my_count, my_map, my_injection
+module Enumerable
+  def my_each
+    return to_enum :my_each unless block_given?
+
+    check_self = is_a?(Range) ? to_a : self
+    i = 0
+    while i < check_self.length
+      yield(check_self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum :my_each_with_index unless block_given?
+
+    i = 0
+    my_each { |x| yield x, i, i += 1 }
+    self
+  end
+
+  def my_select
+    return to_enum :my_select unless block_given?
+
+    arr = []
+    my_each { |x| arr << x if yield(x) }
+    arr
+  end
+
+  def my_all?(param = nil)
+    all = true
+    if block_given?
+      my_each { |x| unless yield(x) then all = false; break end }
+    elsif param
+      my_each { |x| return false unless pattern(x, param) }
+    else
+      my_each { |x| unless x then all = false; break end }
+    end
+    all
+  end
```

Comment:
```ruby
irb(main):006:0> array
=> [3, 4, 7, 6, 0, 8, 6, 0, 1, 3, 2, 0, 0, 5, 6, 2, 0, 7, 7, 2, 2, 8, 6, 8, 3, 5, 5, 0, 6, 2, 5, 6, 2, 3, 1, 2, 8, 5, 5, 7, 7, 7, 7, 5, 7, 5, 8, 2, 7, 4, 5, 2, 7, 2, 3, 1, 6, 3, 8, 4, 8, 6, 8, 3, 0, 4, 8, 2, 1, 4, 4, 3, 2, 2, 6, 8, 6, 7, 5, 7, 4, 6, 1, 2, 5, 1, 5, 4, 4, 4, 2, 0, 4, 2, 7, 4, 4, 5, 1, 5]
irb(main):007:0> array.all?(Integer)
=> true
irb(main):008:0> array.my_all?(Integer)
Traceback (most recent call last):
        5: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        4: from (irb):8
        3: from methods.rb:37:in `my_all?'
        2: from methods.rb:11:in `my_each'
        1: from methods.rb:37:in `block in my_all?'
NoMethodError (undefined method `pattern' for #<Array:0x000055bafc8ffec8>)
Did you mean?  pattern?
irb(main):009:0>
```

I think you wanted to say **pattern?**. With that there is no error.
Please, fix it